shaders/EditorHexGrid.shader

```diff
@@ -0,0 +1,62 @@
+shader_type spatial;
+render_mode unshaded;
+
+// If false the shader will use generated hex patterns for the texture which
+// generaly looks better and uniform but probably slightly slower
+uniform bool useTexture;
+
+uniform sampler2D gridTexture;
+uniform sampler2D maskTexture;
+uniform vec4 color : hint_color;
+
+uniform vec2 gridSize;
+
+varying vec3 worldPos;
+
+const vec2 s = vec2(1.7320508, 1);
+
+// Hexagonal grid pattern code from https://www.shadertoy.com/view/wtdSzX
+
+float hex(in vec2 p){
+    p = abs(p);
+    return max(dot(p, s * .5), p.y);
```

Comment:
If a license is found and this code is kept all floats and doubles need to be in the form of "1.2" or "1.2f"